pkg/apis/core/validation/installation.go

```diff
@@ -66,11 +66,22 @@ func ValidateInstallationDataImports(imports []core.DataImport, fldPath *field.P
 
 	importNames := map[string]bool{}
 	for idx, imp := range imports {
-		if imp.DataRef == "" {
-			allErrs = append(allErrs, field.Required(fldPath.Index(idx).Child("dataRef"), "dataRef must not be empty"))
+		impPath := fldPath.Index(idx)
+
+		if imp.DataRef == "" && imp.SecretRef == nil && imp.ConfigMapRef == nil {
+			allErrs = append(allErrs, field.Required(impPath, "either dataRef, secretRef or configMapRef must not be empty"))
+		}
+
```

Comment:
We should also add a check that _only one_ of `dataRef`, `secretRef`, and `configMapRef` is given.